modin/data_management/data_manager.py

```diff
@@ -233,9 +233,9 @@ def _join_data_manager(self, other, **kwargs):
 
         # This stage is to efficiently get the resulting columns, including the
         # suffixes.
-        self_proxy = pandas.DataFrame(columns=self.columns)
-        other_proxy = pandas.DataFrame(columns=other.columns)
-        new_columns = self_proxy.join(other_proxy, lsuffix=lsuffix, rsuffix=rsuffix).columns
+        self_proxy_columns = pandas.DataFrame(columns=self.columns).columns
+        other_proxy_columns = pandas.DataFrame(columns=other.columns).columns
+        new_columns = self_proxy_columns.append(others_proxy_columns)
```

Comment:
I might have missed this before. This won't end with the correct column names if you don't join. The point here is to get the order of the columns correct and add the correct suffixes to each.